src/com/android/volley/Request.java

```diff
@@ -69,10 +69,13 @@
      * Supported request methods.
      */
     public enum ReturnStrategy {
-        DOUBLE, // You'll get called back into twice, once for cached result and once for network delivery
         NETWORK_IF_NO_CACHE, // You'll get the network response only if the cache misses
         CACHE_IF_NETWORK_FAILS, // You'll get the cached response only if the network errors / fails
-        NETWORK_ONLY // Skip the cache
+        NETWORK_ONLY, // Skip the cache
+        // END LEGACY RETURN STRATEGIES
+        DOUBLE, // You'll get called back into up to twice, once for cached result and once for network delivery
+        SINGLE // guarantees that response will be called only once. exact behavior depends on caching.
```

Comment:
its a combination of NETWORK_IF_NO_CACHE  and CACHE_IF_NETWORK_FAILS. So if it can cache respond without needing a refresh it will do that, and otherwise it will network respond only. It is supposed to eliminate the step of cache respond then network respond.